### ISSUE TITLE
Replace shortcut dropdown menus with textbox that detects keystrokes

### ISSUE THREAD

**Issue by ghost** [fcpxhacks/fcpxhacks#1919]
example: https://drive.google.com/file/d/1XY75Mot8hD1g_SXYyXN1Nj3PmNqw9ADH/view?usp=sharing

Can't you copy the code from an app and just paste it in? I have no idea how coding works

**Comment by latenitefilms** [comment]
Yeah, that's a very good point. We should be able to do something similar. Will investigate.

**Comment by JFtechOfficial** [comment]
The proposed new implementation would remove the ability of setting a shortcut with keys you don't physically have on your keyboard 😢 (I agree that the modifiers dropdown menu is too much tho).
This is my proposal of an ideal UI: not overwhelming, but also not poorer. 
The Select button is removed and the Clear button is shown only when there's the need for it. User can select an action clicking directly on the field (the same behaviour of native MacOS elements).
The space taken by the 2 removed buttons is now dedicated to checkbox that makes easy for the user select and change modifiers.
The last field is a combo box that combines the behaviour of a simple text field with a dropdown menu. It displays a list of values from a pop-up list, but also provides a means for users to type in values. 
It would be cool to have the modifiers checkboxes automatically checked/unchecked when the user types a shortcut in the combo box.


![CP-shortcut-UI](https://user-images.githubusercontent.com/36900161/73352059-98881380-4290-11ea-8ee6-21b213069029.png)


What do you think about it? 🤔

**Comment by latenitefilms** [comment]
@JFtechOfficial - That's a very good point, thanks for sharing! I do really like your proposed UI - it's nice and clean, and really offers the best of both world.

We still have a lot of other things to tick off the to-do list before we tackle this, so we'll have a play with this at some point later in the year. Thanks again!
